datastore/api/snippets_test.py

```diff
@@ -11,291 +11,269 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from functools import wraps
 import time
 
 from gcloud import datastore
+import pytest
 import snippets
-from testing import CloudTest, mark_flaky
+from testing import mark_flaky
 
 
-def eventually_consistent(f):
-    @wraps(f)
-    def inner(self, *args, **kwargs):
-        # This is pretty hacky, but make datastore wait 1s after any
-        # put operation to in order to account for eventual consistency.
-        original_put_multi = self.client.put_multi
+class CleanupClient(datastore.Client):
+    def __init__(self, *args, **kwargs):
+        super(CleanupClient, self).__init__(*args, **kwargs)
+        self.entities_to_delete = []
+        self.keys_to_delete = []
```

Comment:
Could you wrap the relevant methods to automatically add to these lists? I'm thinking something like:

    def record_entity(method, *args, **kwargs):
        entity = method(*args, **kwargs)
        self.entities_to_delete.append(entity)
        return entity
    entity_methods_to_wrap = ('insert', 'lookup', ...)
    for method in entity_methods_to_wrap:
        setattr(self, method, record_entity(getattr(self, method)))

That way if someone ever needs to add a test to this, they don't have to remember to cleanup.

Granted: this is just how the file was doing it before, so don't feel obligated to do this - just a comment.